azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -606,18 +612,21 @@ private void runJob() {
     }
 
     // If the job is still running, set the status to Success.
-    if (!Status.isStatusFinished(node.getStatus())) {
-      changeStatus(Status.SUCCEEDED);
+    if (!Status.isStatusFinished(finalStatus)) {
+      finalStatus = changeStatus(Status.SUCCEEDED);
     }
+    return finalStatus;
   }
 
-  private void changeStatus(Status status) {
+  private Status changeStatus(Status status) {
     changeStatus(status, System.currentTimeMillis());
+    return status;
   }
 
-  private void changeStatus(Status status, long time) {
+  private Status changeStatus(Status status, long time) {
```

Comment:
The caller has the status value already, what's the advantage of returning the same value?